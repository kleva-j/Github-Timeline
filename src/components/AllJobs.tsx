import React from 'react';
import { useQuery } from '@apollo/client';

import { LOAD_JOBS } from '../GraphQL/Queries';
import { Job } from './SingleJob';

export const GetAllJobs: React.FC = () => {
  const { error, loading, data } = useQuery(LOAD_JOBS);

  if (loading) return <p data-testid="loading">Loading...</p>;
  if (error) return <p data-testid="error">Oops, Error occured!!!</p>;

  return (
    <section className="grid__section min-h-screen bg-primary-bg">
      <h2>Github Jobs Listings</h2>
      <article className="flex flex-wrap justify-evenly">
        {data?.jobs.map(({ id, title, type, company, url, location, created_at }: Job) => (
          <aside data-testid={id} key={id} className="bg-primary-input mx-4 my-4">
            <h5 data-testid="title">{title}</h5>
            <p data-testid="company">{company}</p>
            <p data-testid="type">{type}</p>
            <p data-testid="location">{location}</p>
            <p>{created_at}</p>
            <a href={url}>Read More...</a>
          </aside>
        ))}
      </article>
    </section>
  );
};
