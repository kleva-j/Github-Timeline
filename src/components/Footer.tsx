import React from 'react';

interface Props {
  absolute?: boolean;
}

export const Footer: React.FC<Props> = ({ absolute }) => {
  return (
    <>
      <footer
        data-testid="footer"
        className={(absolute ? 'absolute w-full bottom-0 bg-gray-900' : 'relative') + ' pb-6'}
      >
        <div className="container mx-auto px-4">
          <hr className="mb-6 border-b-1 border-gray-700" />
          <div className="flex flex-wrap items-center md:justify-between justify-center">
            <div className="w-full md:w-4/12 px-4">
              <div className="text-sm text-white font-semibold py-1">
                Copyright Â© {new Date().getFullYear()}{' '}
                <a
                  href="http://michaelobasi.com"
                  className="text-white hover:text-gray-400 text-sm font-semibold py-1"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Github Timeline
                </a>
              </div>
            </div>
            <div className="w-full md:w-8/12 px-4">
              <ul className="flex flex-wrap list-none md:justify-end  justify-center">
                <li>
                  <a
                    href="http://michaelobasi.com"
                    className="text-white hover:text-gray-400 text-sm font-semibold block py-1 px-3"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Michael Obasi
                  </a>
                </li>
                <li>
                  <a
                    href="http://michaelobasi.com"
                    className="text-white hover:text-gray-400 text-sm font-semibold block py-1 px-3"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    About Me
                  </a>
                </li>
                <li>
                  <a
                    href="https://frontendtips.netlify.app/"
                    className="text-white hover:text-gray-400 text-sm font-semibold block py-1 px-3"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Blog
                  </a>
                </li>
                <li>
                  <a
                    href="https://github.com/kleva-j/Github-Timeline/blob/develop/LICENCE"
                    className="text-white hover:text-gray-400 text-sm font-semibold block py-1 px-3"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    MIT License
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </footer>
    </>
  );
};
