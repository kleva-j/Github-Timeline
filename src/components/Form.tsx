import React from 'react';

export const SearchForms: React.FC = () => {
  return (
    <form className="flex flex-col sm:flex-row max-w-3xl mx-auto">
      <aside className="flex-1 my-2 md:mx-2 md:ml-0">
        <label htmlFor="desc" className="text-sm font-semibold">Job Description</label>
        <input
          id="desc"
          type="text"
          name="description"
          placeholder="Filter by title, benefits, companies, expertise"
          className="w-full py-2 px-4 border rounded bg-white"
        />
      </aside>

      <aside className="flex-1 my-2 sm:mx-2 md:mr-0">
        <label htmlFor="location" className="text-sm font-semibold">Location</label>
        <input
          type="text"
          id="location"
          name="location"
          placeholder="Filter by city, state, zip code or country"
          className="w-full py-2 px-4 border rounded bg-white"
        />
      </aside>
    </form>
  );
};
