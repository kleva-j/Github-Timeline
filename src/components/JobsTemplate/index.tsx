export * from './template1';
export * from './template2';
