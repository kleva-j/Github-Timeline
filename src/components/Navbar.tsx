/* eslint-disable react/style-prop-object */
import React from 'react';
import { Link } from 'react-router-dom';

import Bars from 'assets/svg/Bars.svg';

interface Props {
  transparent?: boolean;
  navState?: boolean;
}

export const Navbar: React.FC<Props> = ({ transparent, navState = false }) => {
  const [navbarOpen, setNavbarOpen] = React.useState(navState);
  return (
    <nav
      data-testid="navbar"
      className={
        (transparent ? 'top-0 absolute z-50 w-full' : 'relative bg-white shadow-lg') +
        ' flex flex-wrap items-center justify-between px-2 py-3 navbar-expand-lg'
      }
    >
      <div className="container px-4 mx-auto flex flex-wrap items-center justify-between">
        <div className="w-full relative flex justify-between lg:w-auto lg:static lg:block lg:justify-start">
          <Link
            className={
              (transparent ? 'text-white' : 'text-gray-800') +
              ' text-sm font-bold leading-relaxed inline-block mr-4 py-2 whitespace-no-wrap uppercase'
            }
            to="/"
          >
            Github Timeline
          </Link>
          <button
            className="cursor-pointer text-xl leading-none px-3 py-1 border border-solid border-transparent rounded bg-transparent block lg:hidden outline-none focus:outline-none"
            type="button"
            onClick={() => setNavbarOpen(!navbarOpen)}
          >
            <i className={(transparent ? 'text-white' : 'text-gray-800') + ' fas fa-bars'}>
              <img src={Bars} alt="burger-nav" />
            </i>
          </button>
        </div>
        <div
          data-testid="list"
          className={
            'lg:flex flex-grow items-center bg-white lg:bg-transparent lg:shadow-none' +
            (navbarOpen ? ' block rounded shadow-lg' : ' hidden')
          }
          id="example-navbar-warning"
        >
          <ul className="flex flex-col lg:flex-row list-none mr-auto">
            <li className="flex items-center">
              <Link
                className={
                  (transparent
                    ? 'lg:text-white lg:hover:text-gray-300 text-gray-800'
                    : 'text-gray-800 hover:text-gray-600') +
                  ' px-3 py-4 lg:py-2 flex items-center text-xs uppercase font-bold'
                }
                to="/jobs"
              >
                <i
                  className={
                    (transparent ? 'lg:text-gray-300 text-gray-500' : 'text-gray-500') +
                    ' far fa-file-alt text-lg leading-lg mr-2'
                  }
                />{' '}
                Jobs
              </Link>
            </li>
          </ul>
          <ul className="flex flex-col lg:flex-row list-none lg:ml-auto">
            <li className="flex items-center">
              <Link
                className={
                  (transparent
                    ? 'lg:text-white lg:hover:text-gray-300 text-gray-800'
                    : 'text-gray-800 hover:text-gray-600') +
                  ' px-3 py-4 lg:py-2 flex items-center text-xs uppercase font-bold'
                }
                to="#twitter"
              >
                <i
                  className={
                    (transparent ? 'lg:text-gray-300 text-gray-500' : 'text-gray-500') +
                    ' fab fa-twitter text-lg leading-lg '
                  }
                />
                <span className="lg:hidden inline-block ml-2">Tweet</span>
              </Link>
            </li>

            <li className="flex items-center">
              <Link
                className={
                  (transparent
                    ? 'lg:text-white lg:hover:text-gray-300 text-gray-800'
                    : 'text-gray-800 hover:text-gray-600') +
                  ' px-3 py-4 lg:py-2 flex items-center text-xs uppercase font-bold'
                }
                to="#github-stars"
              >
                <i
                  className={
                    (transparent ? 'lg:text-gray-300 text-gray-500' : 'text-gray-500') +
                    ' fab fa-github text-lg leading-lg '
                  }
                />
                <span className="lg:hidden inline-block ml-2">Star</span>
              </Link>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  );
};
