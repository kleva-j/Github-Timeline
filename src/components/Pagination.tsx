import PropTypes from 'prop-types';
import React, { useEffect } from 'react';
import { usePagination } from 'react-pagination-hook';

interface IProps {
  noOfItems: number;
  setPage: Function;
}

export const Pager: React.FC<IProps> = ({ noOfItems=0, setPage }) => {
  const limit = noOfItems / 10;
  const numberOfPages = noOfItems % 10 ? limit + 1 : limit;
  const { activePage, isFirst, isLast, hasPrevious, hasNext, visiblePieces, goToPage } = usePagination({
    initialPage: 1,
    maxButtons: 2,
    numberOfPages,
  });

  useEffect(() => {
    setPage(activePage);
  }, [activePage, setPage]);

  return (
    <nav aria-label="Page navigation" className="flex justify-center my-10">
      <ul className="inline-flex space-x-2 items-center">
        <li>
          <button
            data-testid="prev-button"
            className={`flex items-center justify-center w-10 h-10 text-gray-600 transition-colors duration-150 border rounded focus:outline-none hover:bg-gray-100 ${
              isFirst && 'cursor-not-allowed'
            } `}
            onClick={() => {
              hasPrevious && goToPage(activePage - 1);
              !isFirst && window.scrollTo(0, 0);
            }}
          >
            <svg className="w-4 h-4 fill-current" viewBox="0 0 20 20">
              <path d="M12.707 5.293a1 1 0 010 1.414L9.414 10l3.293 3.293a1 1 0 01-1.414 1.414l-4-4a1 1 0 010-1.414l4-4a1 1 0 011.414 0z"></path>
            </svg>
          </button>
        </li>
        {visiblePieces
          .filter((item) => item.type === 'page-number')
          .map(({ pageNumber }: any, index: number) => (
            <li key={index}>
              <button
                data-testid={`pageNumber-${pageNumber}`}
                className={`w-10 h-10 ${
                  pageNumber === activePage
                    ? 'text-white bg-gray-900 border border-r-0 border-gray-600'
                    : 'text-gray-600 hover:bg-gray-100'
                } transition-colors duration-150 rounded focus:outline-none`}
                onClick={() => {
                  goToPage(pageNumber);
                  window.scrollTo(0, 0);
                }}
              >
                {pageNumber}
              </button>
            </li>
          ))}
        <li>
          <button
            data-testid="next-button"
            className={`flex items-center justify-center w-10 h-10 text-gray-600 transition-colors duration-150 bg-white border rounded focus:outline-none hover:bg-gray-100 ${
              !hasNext && 'cursor-not-allowed'
            } `}
            onClick={() => {
              hasNext && goToPage(activePage + 1);
              !isLast && window.scrollTo(0, 0);
            }}
          >
            <svg className="w-4 h-4 fill-current" viewBox="0 0 20 20">
              <path d="M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z"></path>
            </svg>
          </button>
        </li>
      </ul>
    </nav>
  );
};

Pager.propTypes = {
  noOfItems: PropTypes.number.isRequired,
  setPage: PropTypes.func.isRequired,
};
