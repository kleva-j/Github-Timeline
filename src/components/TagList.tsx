import React, { memo, useCallback, useState } from 'react';
import { Tag, TagContainer } from 'styled';

type tTag = {
  title: string;
  isActive: boolean;
};

const Tags: tTag[] = [
  { title: 'Full Time', isActive: false },
  { title: 'remote', isActive: false },
  { title: 'HTML', isActive: false },
  { title: 'CSS', isActive: false },
  { title: 'SASS', isActive: false },
  { title: 'JavaScript', isActive: false },
  { title: 'TypeScript', isActive: false },
  { title: 'React', isActive: false },
  { title: 'Angular', isActive: false },
  { title: 'Vue', isActive: false },
  { title: 'Svelte', isActive: false },

  { title: 'Java', isActive: false },
  { title: 'Ruby', isActive: false },
  { title: 'Python', isActive: false },
  { title: 'Nodejs', isActive: false },
  { title: 'Go', isActive: false },
  { title: 'Swift', isActive: false },
  { title: 'C', isActive: false },
  { title: 'C++', isActive: false },
  { title: 'C#', isActive: false },
  { title: 'Scala', isActive: false },
  { title: 'R', isActive: false },

  { title: 'Kotlin', isActive: false },
  { title: 'PHP', isActive: false },
  { title: 'Rust', isActive: false },
  { title: 'Elm', isActive: false },
  { title: 'Objective-C', isActive: false },
  { title: 'LISP', isActive: false },
  { title: 'Perl', isActive: false },
  { title: 'SQL', isActive: false },
  { title: 'Haskel', isActive: false },
  { title: 'Erlang', isActive: false },
  { title: 'Clojure', isActive: false },
];

export const TagList: React.FC<{ setParams: Function }> = memo(({ setParams }) => {
  const [state, setState] = useState(Tags);

  const handleClick = (index: number, isActive: boolean) => {
    const newList = [...state];
    newList[index].isActive = !isActive;
    setState(() => newList);
    handleUpdate();
  };

  const handleUpdate = useCallback(() => {
    const newState = state;
    const params = newState
      .filter((item) => item.isActive)
      .reduce(
        (prev: any, next: tTag) => {
          let { description, location, full_time } = prev;
          if (next.title === 'Full Time') full_time = 'on';
          else if (next.title === 'remote') location = 'remote';
          else description = [...description, next.title];
          return { description, location, full_time };
        },
        { description: [], location: '', full_time: '' },
      );
    setParams({
      description: params.description.join('+') || undefined,
      location: params.location || undefined,
      full_time: params.full_time || undefined,
    });
  }, [state, setParams]);

  return (
    <TagContainer>
      {state.map(({ title, isActive }, index) => (
        <Tag data-testid={`id-${index}`} key={title} active={isActive} onClick={() => handleClick(index, isActive)}>
          {title}
        </Tag>
      ))}
    </TagContainer>
  );
});
