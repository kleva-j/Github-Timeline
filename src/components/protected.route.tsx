import React from 'react';
// import { Redirect, Route, RouteComponentProps, RouteProps } from 'react-router-dom';

// import { useAuthState } from 'contexts/AuthContext';

// interface Props extends RouteProps {
//   component: React.ComponentType<RouteComponentProps<any>> | React.ComponentType<any>;
// }

// export const ProtectedRoute: React.FC<Props> = ({ component: Component, ...rest }) => {
//   const { isAuthenticated } = useAuthState();

//   return (
//     <Route
//       {...rest}
//       render={(props) =>
//         isAuthenticated ? (
//           <Component {...props} />
//         ) : (
//           <Redirect
//             to={{
//               pathname: '/login',
//               state: {
//                 from: props.location,
//               },
//             }}
//           />
//         )
//       }
//     />
//   );
// };
