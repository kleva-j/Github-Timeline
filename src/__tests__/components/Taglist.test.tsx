import { cleanup, fireEvent, render } from '@testing-library/react';

import { TagList } from 'components/TagList';

describe('Navbar', () => {
  afterEach(cleanup);

  test('Should render the Taglist component.', () => {
    const { container, getByText } = render(<TagList />);

    expect(container.firstChild).toMatchSnapshot();
    expect(getByText('remote')).toBeTruthy();
    expect(getByText('HTML')).toBeTruthy();
  });

  test('Should render the Taglist component and toggle state.', () => {
    const { getByTestId } = render(<TagList />);

    const tag2 = getByTestId('id-1');
    fireEvent.click(tag2);
    expect(tag2).toHaveStyle({ background: '#1a202c' });
  });
});
