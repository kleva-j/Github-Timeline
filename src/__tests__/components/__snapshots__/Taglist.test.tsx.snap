// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Should render the Taglist component. 1`] = `
<section
  class="sc-jSgupP bpEqdu"
>
  <button
    class="sc-gKsewC erNndT"
    data-testid="id-0"
  >
    Full Time
  </button>
  <button
    class="sc-gKsewC erNndT"
    data-testid="id-1"
  >
    remote
  </button>
  <button
    class="sc-gKsewC erNndT"
    data-testid="id-2"
  >
    HTML
  </button>
  <button
    class="sc-gKsewC erNndT"
    data-testid="id-3"
  >
    CSS
  </button>
  <button
    class="sc-gKsewC erNndT"
    data-testid="id-4"
  >
    SASS
  </button>
  <button
    class="sc-gKsewC erNndT"
    data-testid="id-5"
  >
    JavaScript
  </button>
  <button
    class="sc-gKsewC erNndT"
    data-testid="id-6"
  >
    TypeScript
  </button>
  <button
    class="sc-gKsewC erNndT"
    data-testid="id-7"
  >
    React
  </button>
  <button
    class="sc-gKsewC erNndT"
    data-testid="id-8"
  >
    Angular
  </button>
  <button
    class="sc-gKsewC erNndT"
    data-testid="id-9"
  >
    Vue
  </button>
  <button
    class="sc-gKsewC erNndT"
    data-testid="id-10"
  >
    Svelte
  </button>
  <button
    class="sc-gKsewC erNndT"
    data-testid="id-11"
  >
    Java
  </button>
  <button
    class="sc-gKsewC erNndT"
    data-testid="id-12"
  >
    Ruby
  </button>
  <button
    class="sc-gKsewC erNndT"
    data-testid="id-13"
  >
    Python
  </button>
  <button
    class="sc-gKsewC erNndT"
    data-testid="id-14"
  >
    Nodejs
  </button>
  <button
    class="sc-gKsewC erNndT"
    data-testid="id-15"
  >
    Go
  </button>
  <button
    class="sc-gKsewC erNndT"
    data-testid="id-16"
  >
    Swift
  </button>
  <button
    class="sc-gKsewC erNndT"
    data-testid="id-17"
  >
    C
  </button>
  <button
    class="sc-gKsewC erNndT"
    data-testid="id-18"
  >
    C++
  </button>
  <button
    class="sc-gKsewC erNndT"
    data-testid="id-19"
  >
    C#
  </button>
  <button
    class="sc-gKsewC erNndT"
    data-testid="id-20"
  >
    Scala
  </button>
  <button
    class="sc-gKsewC erNndT"
    data-testid="id-21"
  >
    R
  </button>
  <button
    class="sc-gKsewC erNndT"
    data-testid="id-22"
  >
    Kotlin
  </button>
  <button
    class="sc-gKsewC erNndT"
    data-testid="id-23"
  >
    PHP
  </button>
  <button
    class="sc-gKsewC erNndT"
    data-testid="id-24"
  >
    Rust
  </button>
  <button
    class="sc-gKsewC erNndT"
    data-testid="id-25"
  >
    Elm
  </button>
  <button
    class="sc-gKsewC erNndT"
    data-testid="id-26"
  >
    Objective-C
  </button>
  <button
    class="sc-gKsewC erNndT"
    data-testid="id-27"
  >
    LISP
  </button>
  <button
    class="sc-gKsewC erNndT"
    data-testid="id-28"
  >
    Perl
  </button>
  <button
    class="sc-gKsewC erNndT"
    data-testid="id-29"
  >
    SQL
  </button>
  <button
    class="sc-gKsewC erNndT"
    data-testid="id-30"
  >
    Haskel
  </button>
  <button
    class="sc-gKsewC erNndT"
    data-testid="id-31"
  >
    Erlang
  </button>
  <button
    class="sc-gKsewC erNndT"
    data-testid="id-32"
  >
    Clojure
  </button>
</section>
`;
